plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -410,21 +416,30 @@ private Optional<DataFileInfo> rewriteParquetFile(Path path, FileDeletion deleti
                 int positionCount = page.getPositionCount();
                 int[] retained = new int[positionCount];
                 int[] deleted = new int[positionCount];
+                int[] deletedBecauseOfUpdate = new int[positionCount];
```

Comment:
We created before this change one array with `positionCount` items. Now we create two arrays with `positionCount` items. Is this valid?

Reply:
We created 2 arrays with positionCount:
```
int[] retained = new int[positionCount];
int[] deleted = new int[positionCount];
```
now we create three, this is valid, these arrays are only used to get specific positions from page

Reply:
Why not use `rowsDeletedBecauseOfUpdate.getLongCardinality()` instead?